go/src/koding/klient/uploader/uploader.go

```diff
@@ -0,0 +1,275 @@
+package uploader
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"time"
+
+	"koding/kites/keygen"
+	"koding/klient/storage"
+	"koding/logrotate"
+
+	"github.com/boltdb/bolt"
+	"github.com/koding/kite"
+	"github.com/koding/logging"
+)
+
+var defaultLog = logging.NewCustom("uploader", false)
+
+// Options represents arguments required to create a Uploader value.
+type Options struct {
+	KeygenURL string      // required
+	Kite      *kite.Kite  // required
+	Bucket    string      // required
+	Region    string      // required
+	DB        *bolt.DB    // optional; in-memory store if nil
+	Log       kite.Logger // optional; defaultLog if nil
+}
+
+// Logger is a wrapper for kite.Logger that implements logging.Logger.
```

Comment:
Now I got it on a second read - the kite.Logger interface can be upgraded to logging.Logger interface, since the underlying value implements logging.Logger when created by kite.

Done, I've removed the wrapper in https://github.com/koding/koding/pull/8947/commits/71eef548cb053017890af6c96fce5f2520b6d159.